bson/src/main/org/bson/codecs/pojo/PojoCodecImpl.java

```diff
@@ -48,41 +43,28 @@
     private final CodecRegistry registry;
     private final PropertyCodecRegistry propertyCodecRegistry;
     private final DiscriminatorLookup discriminatorLookup;
-    private final ConcurrentMap<ClassModel<?>, Codec<?>> codecCache;
     private final boolean specialized;
 
     PojoCodecImpl(final ClassModel<T> classModel, final CodecRegistry codecRegistry,
                   final List<PropertyCodecProvider> propertyCodecProviders, final DiscriminatorLookup discriminatorLookup) {
         this.classModel = classModel;
-        this.registry = fromRegistries(fromCodecs(this), codecRegistry);
```

Comment:
Removed this nesting of registries

Reply:
I noticed we did this when I wanted to test the number of registry lookups for classes and wanted to remove the impediment to be able to determine how many times the registry is actually called for lookups.

It has the benefit of simplifying the code / logic.  Codec lookups are cached by the ProvidersCodecRegistry, and there's no need to create a new ProvidersCodecRegistry and cache for each PojoCodec.

Why was it necessary in the first place? 

This was a workaround to handle fully configured PojoCodecs with their type information and not to use the registry to get the raw PojoCodec which is not specialized with type information.  The `NestedGenericHolderFieldWithMultipleTypeParamsModel` test class covers this scenario, the `nested` field contains the fully reified data for the `PropertyWithMultipleTypeParamsModel`. If the PojoCodec went to the codec registry to obtain a `PropertyWithMultipleTypeParamsModel` instance it wouldn't be specialized and would have no concrete types.  So we still need a mechanism to ensure that if the configured PojoCodec is the discriminated class then we should use that.

Check out the `NestedGenericHolderFieldWithMultipleTypeParamsModel` for an example.